core/src/main/java/com/linecorp/armeria/common/encoding/StreamDecoderFactory.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.encoding;
+
+import com.linecorp.armeria.common.HttpResponse;
+
+import io.netty.buffer.ByteBufAllocator;
+
+/**
+ * An interface that constructs a new {@link StreamDecoder} for a given Content-Encoding header value.
+ * A new decoder is valid for the lifetime of an {@link HttpResponse}.
+ */
+public interface StreamDecoderFactory extends com.linecorp.armeria.client.encoding.StreamDecoderFactory {
```

Comment:
Is there any reason that we are doing this instead of
```java
package com.linecorp.armeria.client.encoding;

public interface StreamDecoderFactory extends com.linecorp.armeria.client.encoding.StreamDecoderFactory {
```
It looks weird this new interface extends the deprecated one. 🤔

Reply:
This is @trustin 's suggestion (https://github.com/line/armeria/pull/3087#pullrequestreview-498477790) that `DecodingClient` can accept both new and deprecated without breaking change.

Reply:
Oops, I have pasted the wrong example. I fixed it, please take a look at it again.

> How about making the deprecated StreamDecoderFactory and StreamDecoder extend the new interfaces in the common package

Yeah, I think @trustin meant extending the new one from the deprecated one. 😄

Reply:
> Is there any reason that we are doing this instead of
> 
> ```java
> package com.linecorp.armeria.client.encoding;
> 
> public interface StreamDecoderFactory extends com.linecorp.armeria.common.encoding.StreamDecoderFactory {
> ```
> 
> It looks weird this new interface extends the deprecated one. 🤔

In this relation, `DecodingClient` can't accept new encoder without additional change(e.g update import statement, create new method that can accept new encoder)... 😢